File: leetcode/476. Number Complement/476.java
// https://helloacm.com/how-to-compute-the-number-complement-for-integers/
// https://leetcode.com/problems/number-complement/

class Solution {
    public int findComplement(int num) {
        int r = 0, x = 1;
        while (num > 0) {
            r += (1 - num % 2) * x;
            num /= 2;
            x *= 2;
        }
        return r;
    }
}

File: leetcode/476. Number Complement/476.js
// https://helloacm.com/how-to-compute-the-number-complement-for-integers/
// https://leetcode.com/problems/number-complement/

/**
 * @param {number} num
 * @return {number}
 */
var findComplement = function(num) {
    const binary = num.toString(2).split("");
    return parseInt(binary.map( b => b == '0' ? '1' : '0' ).join(""), 2);
};
